### PR TITLE
Use TestArtifact plugin instead of custom test jar tasks

### PR DESCRIPTION
**Pull request by breskeby**
simplifies build logic and avoids custom creation of test artifacts

### CODE REVIEW

**Review by pugnascotia — APPROVED** [review]

File: x-pack/plugin/sql/qa/jdbc/security/build.gradle [hunk 1]
```diff
@@ -1,29 +1,22 @@
...
-  from sourceSets.test.output
-}
-
-artifacts {
-  testArtifacts testJar
-}
-
 subprojects {
   // Use tests from the root security qa project in subprojects
   configurations.create('testArtifacts')
```
Comment (mark-vieira) [inline x-pack/plugin/sql/qa/jdbc/security/build.gradle:13]: Shouldn't the plugin create this configuration?

File: x-pack/plugin/sql/qa/server/security/build.gradle [hunk 2]
```diff
@@ -24,7 +15,9 @@ subprojects {
 
   dependencies {
     testImplementation project(":x-pack:plugin:core")
-    testArtifacts project(path: mainProject.path, configuration: 'testArtifacts')
+    testArtifacts(testArtifact(project(mainProject.path))) {
+      transitive = false
```
Comment (mark-vieira) [inline x-pack/plugin/sql/qa/server/security/build.gradle:19]: Why do we have to disable transitive dependencies here? Couldn't we do this on the configuration?

File: x-pack/plugin/sql/qa/jdbc/security/build.gradle (hunk 1 above)
Comment (breskeby) [reply x-pack/plugin/sql/qa/jdbc/security/build.gradle:13]: no. this is the consuming project. the plugin is just for providing the artifact. for consuming we only added the `testArtifact` convenience method see https://github.com/elastic/elasticsearch/blob/fd6d966e44170c2e621d0501c9a94b76f62038c7/build.gradle#L57-L74

File: x-pack/plugin/sql/qa/server/security/build.gradle (hunk 2 above)
Comment (breskeby) [reply x-pack/plugin/sql/qa/server/security/build.gradle:19]: true. changed to be done on the configuration

**Review by mark-vieira — APPROVED** [review]
:+1:

**Comment by breskeby** [comment]
@elasticmachine update branch

**Pull request merged**